### PR TITLE
Update chat example to not broadcast close

### PR DESCRIPTION
**Pull request by mlodato517**
**This PR**
1. Updates the commands in `examples/server.rs` to pass the `--features="async-std-runtime"` flag
1. Updates the commands in `examples/echo-server.rs` to pass the `--features="async-std-runtime"` flag
1. Updates the command in `examples/echo-server.rs` to start the `echo-server` example instead of the `server` example
1. Updates the server example to not broadcast `Close` messages

**Why?**
First two are because, when running the commands without the features flag, you get an error:
```
error: target `server` in package `async-tungstenite` requires the features: `async-std-runtime`
Consider enabling them by passing, e.g., `--features="async-std-runtime"`
```
Third one is so the command starts the example that it's written in - probably just fixing a copy-paste typo.
Last one is because, if clients send a `Close` message to the server, it's broadcast to the other clients, closing their connections (potentially) prematurely.

Tracking issue https://github.com/sdroege/async-tungstenite/issues/22

### CODE REVIEW

File: examples/echo-server.rs [hunk 1]
```diff
@@ -2,11 +2,11 @@
 //!
 //! You can test this out by running:
 //!
-//!     cargo run --example server 127.0.0.1:12345
+//!     cargo run --features="async-std-runtime" --example echo-server 127.0.0.1:12345
```
Comment (mlodato517) [inline examples/echo-server.rs:5]: Would you like this command updates split into another PR because they don't have anything to do with the "not broadcast close" thing? I guess I was being a little lazy and just stuffing everything together.

File: examples/server.rs [hunk 2]
```diff
@@ -52,26 +52,28 @@ async fn handle_connection(peer_map: PeerMap, raw_stream: TcpStream, addr: Socke
 
     let (outgoing, incoming) = ws_stream.split();
 
-    let broadcast_incoming = incoming.try_for_each(|msg| {
```
Comment (mlodato517) [inline examples/server.rs:55]: Best viewed with "Hide whitespace changes"

File: examples/echo-server.rs (hunk 1 above)
Comment (sdroege) [reply examples/echo-server.rs:5]: That's ok with me as long as it's a separate commit in the same PR :)

File: examples/server.rs [hunk 3]
```diff
@@ -52,26 +52,28 @@ async fn handle_connection(peer_map: PeerMap, raw_stream: TcpStream, addr: Socke
...
-            .map(|(_, ws_sink)| ws_sink);
-
-        for recp in broadcast_recipients {
-            recp.unbounded_send(msg.clone()).unwrap();
-        }
-
-        future::ok(())
-    });
+    let broadcast_incoming = incoming
+        .try_filter(|msg| future::ready(!msg.is_close()))
```
Comment (mlodato517) [inline examples/server.rs]: Should this have a comment because it's an `examples`? It's hard to sum this up in one line but something like
```suggestion
        .try_filter(|msg| {
          // If we broadcast the Close message from one client,
          // it will close the other connected clients.
          future::ready(!msg.is_close())
        })
```
or something along those lines

Comment (sdroege) [reply examples/server.rs]: Seems like a good idea to me, yes

Comment (mlodato517) [reply examples/server.rs]: https://github.com/sdroege/async-tungstenite/pull/23/commits/2a3c60836c8aebb113d0e6197ba4749ebc8d8b3c

Would you like that squashed before merging?

Comment (sdroege) [reply examples/server.rs]: Yes that would be best. Thanks!

File: examples/server.rs [hunk 4]
```diff
@@ -52,26 +52,32 @@ async fn handle_connection(peer_map: PeerMap, raw_stream: TcpStream, addr: Socke
...
-
-        future::ok(())
-    });
+    let broadcast_incoming = incoming
+        .try_filter(|msg| {
+            // Broadcasting a Close message from one client
+            // will close the other clients.
+            future::ready(!msg.is_close())
+        })
+        .try_for_each(|msg| {
```
Comment (sdroege) [inline examples/server.rs:61]: The same change is also needed for `server.rs` or not?

Comment (mlodato517) [reply examples/server.rs:61]: I think this is `server.rs`? Did you mean `echo-server.rs` or `autobahn-server.rs`? Or a different file?

I didn't put it in `echo-server.rs` because I didn't think that one broadcast messages to other clients but I didn't actually read it.

And I didn't consider `autobahn-server.rs` but definitely can!

Comment (sdroege) [reply examples/server.rs:61]: Indeed, nevermind. I should get another coffee :) All fine, `autobahn-server.rs` should stay as it is.
